### ISSUE TITLE
New Unit Creation

### ISSUE THREAD

**Issue by kheaactua** [kheaactua/diplomacy-engine#7]
> Also, new units can only be created in the home territories in the fall, not everywhere on the map, according to the original game rules. The total number of supply centers worldwide are still taken into account. Furthermore, the creation is voluntary and is specified with a build order.

In the spring, right?

Total number worldwide?  Worldwide for a specific empire you mean, right?

Creation can only occur on unoccupied (but owned) supply centers though, as I understand.

**Comment by gobyrne** [comment]
Creation can only occur on unoccupied, owned AND home supply centres. Apparently there are two classes of supply centres. Note that a "Home" supply centre is only a "Home" for a given empire.

**Comment by kheaactua** [comment]
What is a "Home" supply center?  Is it one that you started with?

**Comment by gobyrne** [comment]
I think it usually is. But Alex's board isn't the usual one, so I'm not sure.

**Comment by kheaactua** [comment]
personally I think this rule is dumb...

**Comment by kheaactua** [comment]
What do you think @bochincha

**Comment by bochincha** [comment]
"Home" as in literally the native territory of the empire: if I'm playing the Netherlands I can only raise an army (of Dutchman) in the territory of the Netherlands (continental Europe). It makes sense in the historical context as you would not expect to build a fleet in Dutch South Africa, for instance.

**Comment by kheaactua** [comment]
Is the Netherlands composed of many territories?  How would I know which is an empire's home territory?  Is it any one they start on initially?

**Comment by kheaactua** [comment]
Is the Netherlands composed of many territories?  How would I know which is an empire's home territory?  Is it any one they start on initially?

**Comment by bochincha** [comment]
Gray's on the right track. Think colonial times. You basically have 2 types of supply centers: the "Home" ones, say Paris [ID 192] for the French, then you have the "Away" ones, like say Ontario [ID 50] or British Columbia [ID 11] for England (i.e. the one that they conquered from the natives). _Both_ of  these count towards the Total number of supply centers, but the player can _only_ raise an armed force in the non-colonial ones. For Empires like USA, CSA, Mexico, Brazil, Russia, China, Turkey etc, I think these are all just the starting supply centers (since these were never European colonial powers).

**Comment by kheaactua** [comment]
Yes, but what is a HOME territory?  Is it an initial territory?  How do I - someone with only territory IDs in front of him, some marked as initially owned by X, and some marked as supply centers - know what a home territory is? :)

**Comment by gobyrne** [comment]
I think we basically need a list of all supply centers with their respective Empire that calls them "Home" (or none if they are not home supplies for anyone).

**Comment by kheaactua** [comment]
If that's the case, I'm tempted to label that as a RELEASE 1.1 milestone..  The British Empire created India and the Indian army.  They then drew supplies from India.  European powers also supplied their forces with Natives in the American Civil War (what we're playing.)  Africans would regularly supply their armies with conquered peoples.  You can totally supply from remote locations.

**Comment by bochincha** [comment]
No fleets though :)

Either way, we have 2 days, so whatever is good enough for Friday is good
enough for me. We can fix/debate this later.
